From 9f95b17b77cf68afb6a02b1eb269957fc6b7b57b Mon Sep 17 00:00:00 2001
From: Pedro Palazon Candel <pedro@joyent.com>
Date: Thu, 26 Oct 2017 16:25:35 +0200
Subject: [PATCH] TOOLS-1902 'sdcadm experimental up -s UUID' should validate
 provided UUIDs better Reviewed by: Trent Mick <trentm@gmail.com> Approved by:
 Trent Mick <trentm@gmail.com>

---
 lib/procedures/index.js | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/lib/procedures/index.js b/lib/procedures/index.js
index 94780c0..893962a 100644
--- a/lib/procedures/index.js
+++ b/lib/procedures/index.js
@@ -692,6 +692,14 @@ function coordinatePlan(opts, cb) {
             // Make sure we get a list of uuids even if hostnames may be
             // initially present as servers option:
             if (opts.servers && opts.servers.length) {
+                var notFound = opts.servers.filter(function (s) {
+                    return (!opts.serverFromUuidOrHostname[s]);
+                });
+                if (notFound.length) {
+                    next(new errors.UpdateError(format(
+                            'unknown servers "%s"', notFound.join('", "'))));
+                    return;
+                }
                 opts.servers = opts.servers.map(function (s) {
                     return opts.serverFromUuidOrHostname[s].uuid;
                 });
-- 
2.21.0

